crates/shared/src/current_block.rs

```diff
@@ -1,325 +1,57 @@
-    }
-
-    async fn block(&self, number: u64) -> Result<BlockNumberHash> {
-        let block = get_block_info_at_id(self, U64::from(number).into()).await?;
-        Ok((block.number, block.hash))
-    }
-
-    /// get blocks defined by the range (inclusive)
-    /// if successful, function guarantees full range of blocks in Result (does
-    /// not return partial results)
-    async fn blocks(&self, range: RangeInclusive<u64>) -> Result<Vec<BlockNumberHash>> {
-        let include_txs = helpers::serialize(&false);
-        let (start, end) = range.into_inner();
-        let mut batch_request = Vec::with_capacity((end - start + 1) as usize);
-        for i in start..=end {
-            let num = helpers::serialize(&BlockNumber::Number(i.into()));
-            let request = self
-                .transport()
-                .prepare("eth_getBlockByNumber", vec![num, include_txs.clone()]);
-            batch_request.push(request);
-        }
-
-        // send_batch guarantees the size and order of the responses to match the
-        // requests
-        self.transport()
-            .send_batch(batch_request.iter().cloned())
-            .await?
-            .into_iter()
-            .map(|response| match response {
-                Ok(response) => {
-                    serde_json::from_value::<web3::types::Block<H256>>(response.clone())
-                        .with_context(|| format!("unexpected response format: {response:?}"))
-                        .and_then(|response| {
-                            Ok((
-                                response.number.context("missing block number")?.as_u64(),
-                                response.hash.context("missing hash")?,
-                            ))
-                        })
-                }
-                Err(err) => Err(anyhow!("web3 error: {}", err)),
-            })
-            .collect()
-    }
-}
-
-async fn get_block_info_at_id(web3: &Web3, id: BlockId) -> Result<BlockInfo> {
-    web3.eth()
-        .block(id)
-        .await
-        .with_context(|| format!("failed to get block for {id:?}"))?
-        .with_context(|| format!("no block for {id:?}"))?
-        .try_into()
-}
-
-pub async fn timestamp_of_block_in_seconds(web3: &Web3, block_number: BlockNumber) -> Result<u32> {
-    Ok(web3
-        .eth()
-        .block(block_number.into())
-        .await
-        .context("failed to get latest block")?
-        .context("block should exists")?
-        .timestamp
-        .as_u32())
-}
-
-pub async fn timestamp_of_current_block_in_seconds(web3: &Web3) -> Result<u32> {
-    timestamp_of_block_in_seconds(web3, BlockNumber::Latest).await
-}
-
-pub async fn block_number_to_block_number_hash(
-    web3: &Web3,
-    block_number: BlockNumber,
-) -> Option<BlockNumberHash> {
-    web3.eth()
-        .block(BlockId::Number(block_number))
+/// Command line arguments for creating global block stream.
+#[derive(Debug, Parser)]
+#[group(skip)]
+pub struct Arguments {
```

Comment:
Maybe move this to `shared::arguments::Arguments`?

Reply:
Initially I wanted to do that together with moving `ethrpc` arguments because I forgot that last time. But I can do that now as well.

Reply:
Follow up is fine.